assignments/arrays.js

```diff
@@ -63,32 +63,49 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+for (const car in inventory) {
+  if (inventory.hasOwnProperty(car)) {
+    const element = inventory[car];
+    if (element.id === 33) {
+      console.log(`Car 33 is a ${element.car_year} ${element.car_make} ${element.car_model}` );
+    }
+  }
+}
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory[inventory.length - 1];
+console.log(`${lastCar.car_make} ${lastCar.car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+let carModels = inventory.sort((a, b) => (a.car_model > b.car_model) ? 1 : -1);
+console.log(carModels);
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
-let carYears = [];
-console.log();
+let carYears = inventory.map(a => a.car_year);
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars = [];
-console.log(); 
+for (const year of carYears) {
+  if (year < 2000) {
+    oldCars.push(year);
+  }
+}
+console.log(oldCars.length); 
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi = [];
-console.log();
+for (const car of inventory) {
+  if (car.car_make === "Audi" || car.car_make === "BMW") {
+    BMWAndAudi.push(car);
+  }
+}
+console.log(BMWAndAudi);
```

Comment:
You forgot to JSON.stringify() your array here. This is just as easy as passing your array in the that JSON method.

If you don't know what JSON is, don't worry you'll learn about it later on (and you'll using it a lot), but if you have time check it out!